packages/studio-base/src/components/DirectTopicStatsUpdater.tsx

```diff
@@ -50,53 +82,107 @@ const selectTopicStats = (ctx: MessagePipelineContext) =>
  * @property interval - the interval, in frames, between updates.
  */
 export function DirectTopicStatsUpdater({ interval = 1 }: { interval?: number }): JSX.Element {
+  const currentTime = useMessagePipeline(selectCurrentTime);
   const startTime = useMessagePipeline(selectStartTime);
   const endTime = useMessagePipeline(selectEndTime);
   const topicStats = useMessagePipeline(selectTopicStats);
+  const playerCapabilities = useMessagePipeline(selectPlayerCapabilities);
+  const playerId = useMessagePipeline(selectPlayerId);
   const duration = endTime && startTime ? subtractTimes(endTime, startTime) : { sec: 0, nsec: 0 };
 
+  const latestCurrentTime = useLatest(currentTime);
   const latestDuration = useLatest(duration);
   const latestStats = useLatest(topicStats);
   const updateCount = useRef(0);
   const rootRef = useRef<HTMLDivElement>(ReactNull);
+  const samples = useRef<Record<string, StatSample>>({});
```

Comment:
I find things like this easier to read if the key is part of the name:
```suggestion
  const samplesByTopic = useRef<Record<string, StatSample>>({});
```